source/opt/passes.cpp

```diff
@@ -71,5 +79,88 @@ bool FreezeSpecConstantValuePass::Process(ir::Module* module) {
   return modified;
 }
 
+bool EliminateDeadConstantPass::Process(ir::Module* module) {
```

Comment:
At some point the passes.cpp file will gain too many responsibilities.  At some point the non-triival passses should be moved  / placed into their own files.